overlord/configstate/config/transaction.go

```diff
@@ -103,6 +104,49 @@ func (t *Transaction) Changes() []string {
 	return out
 }
 
+// shadowsVirtualConfig checks that the given subkeys/value does not
+// "block" the path to a virtual config with a non-map type. E.g. if
+// "network.netplan" is virtual it must be impossible to set
+// "network=false" or getting the document under "network" would be
+// wrong.
+func shadowsVirtualConfig(instanceName string, subkeys []string, value interface{}) error {
+	// maps never block the path
+	if v := reflect.ValueOf(value); v.Kind() == reflect.Map {
+		return nil
+	}
+
+	// helper to compare if two subkeys slices have a common prefix
+	keysHaveCommonPrefix := func(s, prefix []string) bool {
```

Comment:
I'm a bit confused by this function: the name (and the documentation comment) talk about a common prefix, but I don't see any prefix comparison in the implementation; instead, it looks like it's checking if the two arrays are equal.

Reply:
We're dealing with configuration paths (keys) in this package, a config path coming from an input as "foo.bar.baz" becomes an array, e.g. {"foo", "bar", "baz"} and then we are working with those. So in this case we're comparing e.g.  {"foo", "bar", "baz"} and  {"foo", "bar"} - the later would be the prefix of the former.

Reply:
Thanks Pawel, your explanation is exactly right. I wonder if there is a better function name or help text to avoid the confusion that Alberto experienced. Ideas (very) welcome :)

Reply:
Maybe something like `configPathHasPrefix := func(configPath, []string prefix)` + reword the comment, e.g. "helper to compare if configPath slice elements match prefix slice elements`?

Reply:
Or just forget about this function is used for, and name it after what it literally does: something line `stringArrayStartsWith()`?

Reply:
I think I found a way to simplify this

Reply:
eh? Now it's too simple! ;-) :-D